### PR TITLE
Implementing INVALID opcode

### PR DESCRIPTION
**Pull request by smblucker**
These changes implement the invalid opcode; these changes attempt to resolve issue here:

https://github.com/hyperledger/burrow/issues/601

### CODE REVIEW

File: execution/evm/vm.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ var (
 	ErrDataStackUnderflow     = errors.New("Data stack underflow")
 	ErrInvalidContract        = errors.New("Invalid contract")
 	ErrNativeContractCodeCopy = errors.New("Tried to copy native contract code")
+	ErrInvalidOpcode          = errors.New("Invalid opcode")
```
Comment (silasdavis) [inline execution/evm/vm.go]: This has the potential to read as 'an opcode we don't know about', when it means 'abort the execution'. I'd be tempted to suggest: `ErrExecutionAborted = errors.New("Execution aborted")`.

File: execution/evm/vm_test.go [hunk 2]
```diff
@@ -374,6 +374,25 @@ func TestMsgSender(t *testing.T) {
...
+	account2 := newAccount(1, 0, 1)
+
+	var gas uint64 = 100000
+
+	bytecode := MustSplice(PUSH32, 0x72, 0x65, 0x76, 0x65, 0x72, 0x74, 0x20, 0x6D, 0x65, 0x73, 0x73, 0x61,
+		0x67, 0x65, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, PUSH1, 0x00, MSTORE, PUSH1, 0x0E, PUSH1, 0x00, INVALID)
+
+	output, err := ourVm.Call(account1, account2, bytecode, []byte{}, 0, &gas)
+	assert.Error(t, err, "Expected invalid opcode error")
```
Comment (silasdavis) [inline execution/evm/vm_test.go]: Let's `assert.Equal(t, ErrInvalidOpcode, err)` here seeing as we have declared error values.

**Comment by silasdavis** [comment]
Could you also try rebasing on develop I have a feeling that might be cause of integration test failures.

File: execution/evm/vm.go [hunk 3]
```diff
@@ -1078,8 +1082,8 @@ func (vm *VM) call(caller acm.Account, callee acm.MutableAccount, code, input []
 		case STATICCALL, SHL, SHR, SAR, RETURNDATASIZE, RETURNDATACOPY:
 			return nil, fmt.Errorf("%s not yet implemented", op.Name())
 		default:
-			vm.Debugf("(pc) %-3v Invalid opcode %X\n", pc, op)
-			return nil, fmt.Errorf("invalid opcode %X", op)
+			vm.Debugf("(pc) %-3v Unknown opcode %X\n", pc, op)
```
Comment (silasdavis) [inline execution/evm/vm.go]: I prefer this name

**Comment by smblucker** [comment]
@silasdavis Should I always be rebasing against develop? For the other opcodes, I have been rebasing against new-opcodes2 without any issues.

**Comment by silasdavis** [comment]
I didn't realise your merge base was a feature branch here. I've re-merge-based this to develop and this is now good to go. We'll need to make sure we merge changes on newopcodes-* to develop.

**Pull request merged**